kytos/core/buffers.py

```diff
@@ -123,6 +127,39 @@ def full(self):
         return self._queue.sync_q.full()
 
 
+class RateLimitedBuffer(KytosEventBuffer):
+    """
+    Extension of KytosEventBuffer with ratelimiting capabilities.
+    """
+    def __init__(
+        self, *args,
+        strategy: limits.strategies.RateLimiter,
+        limit: limits.RateLimitItem,
+        gen_identifiers: Callable[[KytosEvent], Iterable[Hashable]],
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.strategy = strategy
+        self.limit = limit
+        self.gen_identifiers = gen_identifiers
+
+    def get(self):
+        val = super().get()
+        identifiers = self.limit, *self.gen_identifiers(val)
+        while not self.strategy.hit(*identifiers):
+            window_reset, _ = self.strategy.get_window_stats(*identifiers)
+            time.sleep(window_reset - time.time())
+        return val
+
+    async def aget(self):
+        val = await super().aget()
+        identifiers = self.limit, *self.gen_identifiers(val)
+        while not self.strategy.hit(*identifiers):
+            window_reset, _ = self.strategy.get_window_stats(*identifiers)
+            await asyncio.sleep(window_reset - time.time())
```

Comment:
I haven't looked into the source code yet, but do we have a strong guarantee that it won't result in a non negative number here? I wonder if a misconfiguration of the `limits.RateLimitItemPerSecond` would be caught during the initialization.

Reply:
Added in the guards.